pkg/crinit/aggregated/aggregated_test.go

```diff
@@ -0,0 +1,158 @@
+)
+
+func TestCreateRBACObjects(t *testing.T) {
+	testCases := []struct {
+		desc        string
+		opts        *aggregatedClusterRegistryOptions
+		objExpected bool
+	}{
+		{
+			desc: "should create RBAC options",
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test1",
+					ClusterRegistryNamespace: "cr-namespace",
+					DryRun: false,
+				},
+			},
+			objExpected: true,
+		},
+		{
+			desc: "dry run should not create RBAC options",
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test2",
+					ClusterRegistryNamespace: "dry-namespace",
+					DryRun: true,
+				},
+			},
+			objExpected: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			buffer := &bytes.Buffer{}
+			client := fake.NewSimpleClientset()
+			rbacObj, err := createRBACObjects(buffer, client, tc.opts)
+			if err != nil {
+				t.Error("Failed to create an RBAC Object")
+			}
+			createdRBACObj, _ := client.CoreV1().ServiceAccounts(tc.opts.ClusterRegistryNamespace).Get(rbacObj.Name, metav1.GetOptions{})
+			if tc.objExpected {
+				if createdRBACObj == nil {
+					t.Errorf("Failed to create RBAC Object")
+				}
+			} else {
+				if createdRBACObj != nil {
+					t.Errorf("Expected no RBAC Object but got: %v", createdRBACObj)
+				}
+			}
+		})
+	}
+}
+
+func TestCreateAPIService(t *testing.T) {
+	testCases := []struct {
+		desc        string
+		opts        *aggregatedClusterRegistryOptions
+		svcExpected bool
+	}{
+		{
+			desc: "should create an API Service",
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test1",
+					ClusterRegistryNamespace: "cr-namespace",
+					DryRun: false,
+				},
+			},
+			svcExpected: true,
+		},
+		{
+			desc: "should create an API Service",
+			opts: &aggregatedClusterRegistryOptions{
+				SubcommandOptions: options.SubcommandOptions{
+					Name: "test2",
+					ClusterRegistryNamespace: "dry-namespace",
+					DryRun: true,
+				},
+			},
+			svcExpected: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			buffer := &bytes.Buffer{}
+			//ips := []string{"1.2.3.4"}
+			ca, err := triple.NewCA(tc.opts.Name)
+			if err != nil {
+				t.Error("unable to create credential CA")
+			}
+			apisvcClient := fakeclientset.NewSimpleClientset()
+			apisvcObj, err := createAPIService(buffer, apisvcClient, tc.opts, ca.Cert)
+			if err != nil {
+				t.Errorf("unexpected error creating API Service: got %v", err)
+			}
+			apiService, _ := apisvcClient.ApiregistrationV1beta1().APIServices().Get(apisvcObj.Name, metav1.GetOptions{})
+			if tc.svcExpected {
+				if apiService == nil {
+					t.Errorf("Error in creating API service")
+				} else {
+					if apiService.ObjectMeta.Name != apiSvcName {
+						t.Errorf("Unexpected API Service Created. Expected %v, got %v", apiSvcName, apiService.ObjectMeta.Name)
```

Comment:
nit: Unexpected APIService created.

Reply:
I'm not sure which nit you are indicating. Is it that APIService should have no space? Or that I should remove the following sentence (Expected. . .)? Or that the subsequent error messages have left out Service? I have assumed the latter but am not sure if this what you meant. In any case, that needed to be corrected.

Reply:
Yes, it was that the following error messages left out `Service`. Apologies, I should have made the comment on one of those.